tools/autograd/gen_autograd_functions.py

```diff
@@ -52,7 +52,7 @@
 struct TORCH_API ${op} : public ${superclass} {
 #endif
   using ${superclass}::${superclass};
-  variable_list apply(variable_list&& grads) override;
+  variable_list apply(variable_list&& grads, std::optional<PyObject*> compiler = std::nullopt) override;
```

Comment:
PyNode::apply_with_saved calls into PyNode::apply (which calls the backward). There's a lot of shared code, do you think we should fork the implementations with helpers?

Reply:
Just refactor the one file `python_function.cpp` rather than changing the entire codebase.

In the case of `python_function.cpp` I actually don't think apply_with_saved should call 
apply, they need to do very different tings.  We should just implement a different `apply_with_saved` that inserts the right node in the output graph.